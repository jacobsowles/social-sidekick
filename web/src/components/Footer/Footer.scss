@import '@styles/_variables.scss';

.footer {
  background-color: $gray-dark;
  position: absolute;
  color: #fff;
  bottom: 0;
  width: 100%;
  height: 45px;
  line-height: 45px;

  a {
    color: #ffffff;

    &:hover {
      text-decoration: none;
    }
  }

  svg {
    font-size: 1.7em;
    margin-right: 10px;
    vertical-align: -0.25em;
  }
}
