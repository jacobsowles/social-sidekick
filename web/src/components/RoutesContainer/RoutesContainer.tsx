import PropTypes from 'prop-types';
import React, { Component, ComponentState } from 'react';
import { InjectedAlertProp, withAlert } from 'react-alert';
import { connect } from 'react-redux';
import { Action, Dispatch } from 'redux';

import { setUserServicesState } from '@actions/service.actions';
import ApiService from '@core/api';
import AuthService from '@core/auth';
import ServiceModuleFactory from '@core/services';
import { AppState, UserService } from '@core/types';
import Routes from './Routes';

interface RoutesContainerOwnProps {
  alert: InjectedAlertProp;
}

interface RoutesContainerDispatchProps {
  getUserServices: (userId: string) => Promise<UserService[]>;
}

interface RoutesContainerStateProps {
  services: UserService[] | null;
  userId?: string;
}

type RoutesContainerProps = RoutesContainerOwnProps &
  RoutesContainerDispatchProps &
  RoutesContainerStateProps;

class RoutesContainer extends Component<RoutesContainerProps> {
  public static contextTypes = {
    router: PropTypes.object
  };

  private authService = new AuthService();

  public componentDidUpdate(previousProps: RoutesContainerProps) {
    if (this.props.userId && (!this.props.services || this.props.services.length === 0)) {
      this.props.getUserServices(this.props.userId);
    }
  }

  public handleAuthentication = (nextState: ComponentState): void => {
    if (/access_token|id_token|error/.test(nextState.location.hash)) {
      this.authService.handleAuthentication((location: string) =>
        this.context.router.history.replace(location)
      );
    }
  };

  public render() {
    const serviceModules = this.props.services
      ? this.props.services
          .filter(service => service.isConnected)
          .map(service => ServiceModuleFactory.getServiceModule(service.name, this.props.userId))
      : null;

    return (
      <Routes
        handleAuthentication={this.handleAuthentication}
        serviceModules={serviceModules}
        services={this.props.services}
        userId={this.props.userId}
      />
    );
  }
}

const mapDispatchToProps = (
  dispatch: Dispatch<Action>,
  ownProps: RoutesContainerOwnProps
): RoutesContainerDispatchProps => {
  return {
    getUserServices: async (userId: string): Promise<UserService[]> => {
      try {
        const api = new ApiService();
        const response = await api.getUserServices(userId);
        dispatch(setUserServicesState(response.data));
        return response.data;
      } catch (error) {
        ownProps.alert.error('Unable to load list of services.');
      }
    }
  };
};

const mapStateToProps = (state: AppState) => {
  return {
    services: state.services,
    userId: state.user ? state.user.sub : undefined
  };
};

export default withAlert(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(RoutesContainer)
);
