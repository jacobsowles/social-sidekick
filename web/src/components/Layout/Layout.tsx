import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Auth0UserProfile } from 'auth0-js';
import moment from 'moment';
import React, { FunctionComponent } from 'react';

import Footer from '@components/Footer';
import Navbar from '@components/Navbar';
import Routes from '@components/Routes';
import './Layout.scss';

export interface LayoutProps {
  handleLogout: () => void;
  user?: Auth0UserProfile;
}

const Layout: FunctionComponent<LayoutProps> = ({ handleLogout, user }) => {
  console.log('rendering Layout');
  return (
    <div className="layout">
      <Navbar
        handleLogout={handleLogout}
        picture={user ? user.picture : undefined}
        username={user ? user.name : undefined}
      />
      <Routes />
      <Footer>
        <Footer.Content>&copy; Social Sidekick {moment().year()}</Footer.Content>
        <Footer.Content pullRight>
          <a href="https://github.com/jacobsowles/social-sidekick">
            <FontAwesomeIcon icon={['fab', 'github']} title="This project on GitHub" />
            This project on GitHub
          </a>
        </Footer.Content>
      </Footer>
    </div>
  );
};

Layout.defaultProps = {
  user: undefined
};

export default Layout;
