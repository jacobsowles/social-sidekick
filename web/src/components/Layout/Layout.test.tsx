import React from 'react';

import Footer from '@components/Footer';
import RoutesContainer from '@components/RoutesContainer';
import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import Layout, { LayoutProps } from './Layout';

describe('Layout', () => {
  let props: LayoutProps;
  init(() => <Layout {...props} />, 'div');

  beforeEach(() => {
    props = {
      handleLogout: jest.fn(),
      user: undefined
    };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  it('should always render a Router component', () => {
    shallow()
      .find(RoutesContainer)
      .should.have.lengthOf(1);
  });

  it('should always render a footer element', () => {
    shallow()
      .find(Footer)
      .should.have.lengthOf(1);
  });

  it('should only include the default class names', () => {
    shallowTopLevelElement()
      .prop('className')
      .should.equal('layout');
  });
});
