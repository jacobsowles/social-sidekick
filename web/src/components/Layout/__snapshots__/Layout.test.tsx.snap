// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the component without crashing 1`] = `
<div
  className="layout"
>
  <Navbar
    handleLogout={[MockFunction]}
  />
  <WithAlert(Connect(RoutesContainer)) />
  <Footer
    className=""
  >
    <FooterContent
      pullRight={false}
    >
      Â© Social Sidekick 
      2018
    </FooterContent>
    <FooterContent
      pullRight={true}
    >
      <a
        href="https://github.com/jacobsowles/social-sidekick"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "github",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title="This project on GitHub"
          transform={null}
        />
        This project on GitHub
      </a>
    </FooterContent>
  </Footer>
</div>
`;
